SDCTA/src/components/auth/LogInPage.tsx

```diff
@@ -0,0 +1,166 @@
+import { SetStateAction, useState } from 'react';
+import { signInWithEmailAndPassword } from "firebase/auth";
+import './auth.css';
+import { auth } from "../../firebase-config";
+import {logInErrorHandler} from "../../error_handling/auth-errors"
+import {useNavigate} from 'react-router-dom';
+
+import { GoogleAuthProvider, getAuth } from "firebase/auth";
+import { signInWithRedirect, getRedirectResult} from "firebase/auth";
+
+
+export const LogInPage = () => {
+    const [loginEmail, setLoginEmail] = useState("");
+    const [loginPassword, setLoginPassword] = useState("");
+    const [rememberUser,setRememberUser] = useState(false);
+    const navigate = useNavigate();
+
+    const [emailError, setEmailError] = useState("");
+    const [passwordError, setPasswordError] = useState("");
+    const [unknownError, setUnknownError] = useState("");
+
+    const login = async () => {
+        try {
```

Comment:
So I actually think its ok to not tell them about a potential error in their password if their email is wrong. In other words you can just use the firebase error checking. 
But maybe a slightly more concise way to implement this would be to define an InputError type that is an object keyed with error type. So the keys would be strings like email, password, unknown . Then, instead of three states for the different errors you just have one state of type InputError that holds all the errors. Now, When you do this check you don't need if statements. You can just set the error with the key that matches the value of errorMessage[0] since your keys have the same name.